src/gravel/controllers/user.py

```diff
@@ -0,0 +1,74 @@
+# project aquarium's backend
+# Copyright (C) 2021 SUSE, LLC.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+import bcrypt
+import json
+
+from pydantic import BaseModel, Field
+from typing import List, Optional
+
+from gravel.controllers.kv import KV
+
+
+class CredentialsModel(BaseModel):
+	username: str = Field("", title="The user name")
+	password: str = Field("", title="The password in plain text")
+
+
+class UserModel(BaseModel):
+	username: str = Field("", title="The user name")
+	password: str = Field("", title="The password")
+	full_name: str = Field("", title="The full name of the user")
+	disabled: Optional[bool] = Field(False, title="Is the user disabled?")
+
+	def hash_password(self) -> None:
+		salt = bcrypt.gensalt()
+		self.password = bcrypt.hashpw(self.password.encode('utf8'), salt).decode('utf8')
+
+	def verify_password(self, password) -> bool:
+		result = bcrypt.checkpw(password, self.password)
+		return result
+
+
+class UserMgr:
+	def __init__(self, kv: KV):
+		self._kv = kv
+
+	async def enumerate(self) -> List[UserModel]:
+		values = await self._kv.get_prefix('/user/')
+		return [UserModel.parse_obj(json.loads(value)) for value in values]
```

Comment:
This could be a `parse_raw()` instead of having to do a `json.loads()` and a `parse_obj()`, I think.

Reply:
> This could be a parse_raw() instead of having to do a json.loads() and a parse_obj(), I think.

Right, according to https://pydantic-docs.helpmanual.io/usage/models/#helper-functions this is correct.